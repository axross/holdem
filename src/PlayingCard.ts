enum PlayingCard {
  aceSpade = "As",
  deuceSpade = "2s",
  threeSpade = "3s",
  fourSpade = "4s",
  fiveSpade = "5s",
  sixSpade = "6s",
  sevenSpade = "7s",
  eightSpade = "8s",
  nineSpade = "9s",
  tenSpade = "Ts",
  jackSpade = "Js",
  queenSpade = "Qs",
  kingSpade = "Ks",
  aceHeart = "Ah",
  deuceHeart = "2h",
  threeHeart = "3h",
  fourHeart = "4h",
  fiveHeart = "5h",
  sixHeart = "6h",
  sevenHeart = "7h",
  eightHeart = "8h",
  nineHeart = "9h",
  tenHeart = "Th",
  jackHeart = "Jh",
  queenHeart = "Qh",
  kingHeart = "Kh",
  aceDiamond = "Ad",
  deuceDiamond = "2d",
  threeDiamond = "3d",
  fourDiamond = "4d",
  fiveDiamond = "5d",
  sixDiamond = "6d",
  sevenDiamond = "7d",
  eightDiamond = "8d",
  nineDiamond = "9d",
  tenDiamond = "Td",
  jackDiamond = "Jd",
  queenDiamond = "Qd",
  kingDiamond = "Kd",
  aceClub = "Ac",
  deuceClub = "2c",
  threeClub = "3c",
  fourClub = "4c",
  fiveClub = "5c",
  sixClub = "6c",
  sevenClub = "7c",
  eightClub = "8c",
  nineClub = "9c",
  tenClub = "Tc",
  jackClub = "Jc",
  queenClub = "Qc",
  kingClub = "Kc",
}

export default PlayingCard;
