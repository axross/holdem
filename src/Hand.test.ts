import { describe, expect, it, jest, test } from "@jest/globals";
import Hand, { HandActionType } from "./Hand";
import PlayingCard from "./PlayingCard";

describe("Hand", () => {
  describe("#snapshots", () => {
    it("matches the snapshot #1", () => {
      const hand = new Hand({
        playerInitialStackSizes: new Map(
          [105.88, 76, 97.2, 93, 79.6, 66.2].map((v, k) => [k, v])
        ),
        playerCards: new Map([
          [1, [PlayingCard.queenDiamond, PlayingCard.sevenDiamond]],
        ]),
        smallBlindSize: 0.4,
        antiSize: 0,
        communityCards: [],
        preflopActions: [
          { type: HandActionType.fold, playerIndex: 2, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 3, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 4, betSize: 0 },
          { type: HandActionType.raise, playerIndex: 5, betSize: 4 },
          { type: HandActionType.fold, playerIndex: 0, betSize: 0.4 },
          { type: HandActionType.fold, playerIndex: 1, betSize: 1 },
        ],
        flopActions: [],
        turnActions: [],
        riverActions: [],
      });

      expect(hand.snapshots).toMatchSnapshot();
    });

    it("matches the snapshot #2", () => {
      const hand = new Hand({
        playerInitialStackSizes: new Map(
          [82, 103.2, 107.6, 113.2, 100.6, 94.8].map((v, k) => [k, v])
        ),
        playerCards: new Map([
          [2, [PlayingCard.aceDiamond, PlayingCard.kingHeart]],
          [3, [PlayingCard.fourSpade, PlayingCard.fourHeart]],
        ]),
        smallBlindSize: 0.4,
        antiSize: 0,
        communityCards: [
          PlayingCard.fourClub,
          PlayingCard.sevenClub,
          PlayingCard.sixSpade,
          PlayingCard.threeSpade,
        ],
        preflopActions: [
          { type: HandActionType.raise, playerIndex: 2, betSize: 3 },
          { type: HandActionType.call, playerIndex: 3, betSize: 3 },
          { type: HandActionType.fold, playerIndex: 4, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 5, betSize: 0 },
          { type: HandActionType.call, playerIndex: 0, betSize: 3 },
          { type: HandActionType.fold, playerIndex: 1, betSize: 1 },
        ],
        flopActions: [
          { type: HandActionType.check, playerIndex: 0, betSize: 0 },
          { type: HandActionType.bet, playerIndex: 2, betSize: 4.8 },
          { type: HandActionType.call, playerIndex: 3, betSize: 4.8 },
          { type: HandActionType.fold, playerIndex: 0, betSize: 0 },
        ],
        turnActions: [
          { type: HandActionType.bet, playerIndex: 2, betSize: 9.4 },
          { type: HandActionType.raise, playerIndex: 3, betSize: 22 },
          { type: HandActionType.fold, playerIndex: 2, betSize: 9.4 },
        ],
        riverActions: [],
      });

      expect(hand.snapshots).toMatchSnapshot();
    });

    it("calls #buildSnapshots() in the first time it's referred", () => {
      const hand = new Hand({
        playerInitialStackSizes: new Map(
          [105.88, 76, 97.2, 93, 79.6, 66.2].map((v, k) => [k, v])
        ),
        playerCards: new Map([
          [1, [PlayingCard.queenDiamond, PlayingCard.sevenDiamond]],
        ]),
        smallBlindSize: 0.4,
        antiSize: 0,
        communityCards: [],
        preflopActions: [
          { type: HandActionType.fold, playerIndex: 2, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 3, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 4, betSize: 0 },
          { type: HandActionType.raise, playerIndex: 5, betSize: 4 },
          { type: HandActionType.fold, playerIndex: 0, betSize: 0.4 },
          { type: HandActionType.fold, playerIndex: 1, betSize: 1 },
        ],
        flopActions: [],
        turnActions: [],
        riverActions: [],
      });

      const buildSnapshots = jest.spyOn(hand as any, "buildSnapshots");

      expect(buildSnapshots).not.toHaveBeenCalled();

      hand.snapshots;

      expect(buildSnapshots).toHaveBeenCalledTimes(1);
    });

    it("calls #buildSnapshots() only one time", () => {
      const hand = new Hand({
        playerInitialStackSizes: new Map(
          [105.88, 76, 97.2, 93, 79.6, 66.2].map((v, k) => [k, v])
        ),
        playerCards: new Map([
          [1, [PlayingCard.queenDiamond, PlayingCard.sevenDiamond]],
        ]),
        smallBlindSize: 0.4,
        antiSize: 0,
        communityCards: [],
        preflopActions: [
          { type: HandActionType.fold, playerIndex: 2, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 3, betSize: 0 },
          { type: HandActionType.fold, playerIndex: 4, betSize: 0 },
          { type: HandActionType.raise, playerIndex: 5, betSize: 4 },
          { type: HandActionType.fold, playerIndex: 0, betSize: 0.4 },
          { type: HandActionType.fold, playerIndex: 1, betSize: 1 },
        ],
        flopActions: [],
        turnActions: [],
        riverActions: [],
      });

      const buildSnapshots = jest.spyOn(hand as any, "buildSnapshots");

      hand.snapshots;
      hand.snapshots;
      hand.snapshots;

      expect(buildSnapshots).toHaveBeenCalledTimes(1);
    });
  });

  describe("#finalPotSize", () => {
    describe("returns the total bet size accumurated in the hand", () => {
      test("#1", () => {
        const hand = new Hand({
          playerInitialStackSizes: new Map(
            [105.88, 76, 97.2, 93, 79.6, 66.2].map((v, k) => [k, v])
          ),
          playerCards: new Map([
            [1, [PlayingCard.queenDiamond, PlayingCard.sevenDiamond]],
          ]),
          smallBlindSize: 0.4,
          antiSize: 0,
          communityCards: [],
          preflopActions: [
            { type: HandActionType.fold, playerIndex: 2, betSize: 0 },
            { type: HandActionType.fold, playerIndex: 3, betSize: 0 },
            { type: HandActionType.fold, playerIndex: 4, betSize: 0 },
            { type: HandActionType.raise, playerIndex: 5, betSize: 4 },
            { type: HandActionType.fold, playerIndex: 0, betSize: 0.4 },
            { type: HandActionType.fold, playerIndex: 1, betSize: 1 },
          ],
          flopActions: [],
          turnActions: [],
          riverActions: [],
        });

        expect(hand.finalPotSize).toBe(5.4);
      });

      test("#2", () => {
        const hand = new Hand({
          playerInitialStackSizes: new Map(
            [82, 103.2, 107.6, 113.2, 100.6, 94.8].map((v, k) => [k, v])
          ),
          playerCards: new Map([
            [2, [PlayingCard.aceDiamond, PlayingCard.kingHeart]],
            [3, [PlayingCard.fourSpade, PlayingCard.fourHeart]],
          ]),
          smallBlindSize: 0.4,
          antiSize: 0,
          communityCards: [
            PlayingCard.fourClub,
            PlayingCard.sevenClub,
            PlayingCard.sixSpade,
            PlayingCard.threeSpade,
          ],
          preflopActions: [
            { type: HandActionType.raise, playerIndex: 2, betSize: 3 },
            { type: HandActionType.call, playerIndex: 3, betSize: 3 },
            { type: HandActionType.fold, playerIndex: 4, betSize: 0 },
            { type: HandActionType.fold, playerIndex: 5, betSize: 0 },
            { type: HandActionType.call, playerIndex: 0, betSize: 3 },
            { type: HandActionType.fold, playerIndex: 1, betSize: 1 },
          ],
          flopActions: [
            { type: HandActionType.check, playerIndex: 0, betSize: 0 },
            { type: HandActionType.bet, playerIndex: 2, betSize: 4.8 },
            { type: HandActionType.call, playerIndex: 3, betSize: 4.8 },
            { type: HandActionType.fold, playerIndex: 0, betSize: 0 },
          ],
          turnActions: [
            { type: HandActionType.bet, playerIndex: 2, betSize: 9.4 },
            { type: HandActionType.raise, playerIndex: 3, betSize: 22 },
            { type: HandActionType.fold, playerIndex: 2, betSize: 9.4 },
          ],
          riverActions: [],
        });

        expect(hand.finalPotSize).toBe(51);
      });
    });
  });
});
