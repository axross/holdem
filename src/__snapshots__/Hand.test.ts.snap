// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hand #snapshots matches the snapshot #1 1`] = `
Array [
  Object {
    "actionPlayerIndex": null,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {},
    "playerStackSizes": Immutable.Map {
      0: 105.88,
      1: 76,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 66.2,
    },
    "potSize": 0,
    "street": "BEGINNING",
  },
  Object {
    "actionPlayerIndex": 0,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 76,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 66.2,
    },
    "potSize": 0.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 1,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 66.2,
    },
    "potSize": 1.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 2,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      5,
      3,
      4,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 66.2,
    },
    "potSize": 1.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 3,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      5,
      4,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      3: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 66.2,
    },
    "potSize": 1.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 4,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      3: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 66.2,
    },
    "potSize": 1.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 5,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      3: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      5: Object {
        "betSize": 4,
        "type": "RAISE",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 62.2,
    },
    "potSize": 5.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 0,
    "activePlayerIndexes": Immutable.Set [
      5,
      1,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FOLD",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      3: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      5: Object {
        "betSize": 4,
        "type": "RAISE",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 62.2,
    },
    "potSize": 5.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 1,
    "activePlayerIndexes": Immutable.Set [
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FOLD",
      },
      1: Object {
        "betSize": 1,
        "type": "FOLD",
      },
      2: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      3: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      5: Object {
        "betSize": 4,
        "type": "RAISE",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 62.2,
    },
    "potSize": 5.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 5,
    "activePlayerIndexes": Immutable.Set [
      5,
    ],
    "playerLastActions": Immutable.Map {},
    "playerStackSizes": Immutable.Map {
      0: 105.47999999999999,
      1: 75,
      2: 97.2,
      3: 93,
      4: 79.6,
      5: 67.60000000000001,
    },
    "potSize": 0,
    "street": "PREFLOP",
  },
]
`;

exports[`Hand #snapshots matches the snapshot #2 1`] = `
Array [
  Object {
    "actionPlayerIndex": null,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {},
    "playerStackSizes": Immutable.Map {
      0: 82,
      1: 103.2,
      2: 107.6,
      3: 113.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 0,
    "street": "BEGINNING",
  },
  Object {
    "actionPlayerIndex": 0,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 81.6,
      1: 103.2,
      2: 107.6,
      3: 113.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 0.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 1,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 81.6,
      1: 102.2,
      2: 107.6,
      3: 113.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 1.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 2,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 3,
        "type": "RAISE",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 81.6,
      1: 102.2,
      2: 104.6,
      3: 113.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 4.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 3,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      4,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 3,
        "type": "RAISE",
      },
      3: Object {
        "betSize": 3,
        "type": "CALL",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 81.6,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 7.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 4,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
      5,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 3,
        "type": "RAISE",
      },
      3: Object {
        "betSize": 3,
        "type": "CALL",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 81.6,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 7.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 5,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0.4,
        "type": "FORCED_BET",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 3,
        "type": "RAISE",
      },
      3: Object {
        "betSize": 3,
        "type": "CALL",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      5: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 81.6,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 7.4,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 0,
    "activePlayerIndexes": Immutable.Set [
      0,
      1,
      2,
      3,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 3,
        "type": "CALL",
      },
      1: Object {
        "betSize": 1,
        "type": "FORCED_BET",
      },
      2: Object {
        "betSize": 3,
        "type": "RAISE",
      },
      3: Object {
        "betSize": 3,
        "type": "CALL",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      5: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 10,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": 1,
    "activePlayerIndexes": Immutable.Set [
      0,
      3,
      2,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 3,
        "type": "CALL",
      },
      1: Object {
        "betSize": 1,
        "type": "FOLD",
      },
      2: Object {
        "betSize": 3,
        "type": "RAISE",
      },
      3: Object {
        "betSize": 3,
        "type": "CALL",
      },
      4: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      5: Object {
        "betSize": 0,
        "type": "FOLD",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 10,
    "street": "PREFLOP",
  },
  Object {
    "actionPlayerIndex": null,
    "activePlayerIndexes": Immutable.Set [
      0,
      3,
      2,
    ],
    "playerLastActions": Immutable.Map {},
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 10,
    "street": "FLOP",
  },
  Object {
    "actionPlayerIndex": 0,
    "activePlayerIndexes": Immutable.Set [
      0,
      3,
      2,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0,
        "type": "CHECK",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 104.6,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 10,
    "street": "FLOP",
  },
  Object {
    "actionPlayerIndex": 2,
    "activePlayerIndexes": Immutable.Set [
      0,
      3,
      2,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0,
        "type": "CHECK",
      },
      2: Object {
        "betSize": 4.8,
        "type": "BET",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 99.8,
      3: 110.2,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 14.8,
    "street": "FLOP",
  },
  Object {
    "actionPlayerIndex": 3,
    "activePlayerIndexes": Immutable.Set [
      0,
      3,
      2,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0,
        "type": "CHECK",
      },
      2: Object {
        "betSize": 4.8,
        "type": "BET",
      },
      3: Object {
        "betSize": 4.8,
        "type": "CALL",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 99.8,
      3: 105.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 19.6,
    "street": "FLOP",
  },
  Object {
    "actionPlayerIndex": 0,
    "activePlayerIndexes": Immutable.Set [
      2,
      3,
    ],
    "playerLastActions": Immutable.Map {
      0: Object {
        "betSize": 0,
        "type": "FOLD",
      },
      2: Object {
        "betSize": 4.8,
        "type": "BET",
      },
      3: Object {
        "betSize": 4.8,
        "type": "CALL",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 99.8,
      3: 105.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 19.6,
    "street": "FLOP",
  },
  Object {
    "actionPlayerIndex": null,
    "activePlayerIndexes": Immutable.Set [
      2,
      3,
    ],
    "playerLastActions": Immutable.Map {},
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 99.8,
      3: 105.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 19.6,
    "street": "FLOP",
  },
  Object {
    "actionPlayerIndex": 2,
    "activePlayerIndexes": Immutable.Set [
      2,
      3,
    ],
    "playerLastActions": Immutable.Map {
      2: Object {
        "betSize": 9.4,
        "type": "BET",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 90.39999999999999,
      3: 105.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 29,
    "street": "TURN",
  },
  Object {
    "actionPlayerIndex": 3,
    "activePlayerIndexes": Immutable.Set [
      2,
      3,
    ],
    "playerLastActions": Immutable.Map {
      2: Object {
        "betSize": 9.4,
        "type": "BET",
      },
      3: Object {
        "betSize": 22,
        "type": "RAISE",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 90.39999999999999,
      3: 83.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 51,
    "street": "TURN",
  },
  Object {
    "actionPlayerIndex": 2,
    "activePlayerIndexes": Immutable.Set [
      3,
    ],
    "playerLastActions": Immutable.Map {
      2: Object {
        "betSize": 9.4,
        "type": "FOLD",
      },
      3: Object {
        "betSize": 22,
        "type": "RAISE",
      },
    },
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 90.39999999999999,
      3: 83.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 51,
    "street": "TURN",
  },
  Object {
    "actionPlayerIndex": 3,
    "activePlayerIndexes": Immutable.Set [
      3,
    ],
    "playerLastActions": Immutable.Map {},
    "playerStackSizes": Immutable.Map {
      0: 79,
      1: 102.2,
      2: 90.39999999999999,
      3: 134.4,
      4: 100.6,
      5: 94.8,
    },
    "potSize": 0,
    "street": "TURN",
  },
]
`;
